### PR TITLE
consensus: don't use arith_uint256 in consensus.h

### PR DESCRIPTION
**Pull request by theuni**
Requiring arith_uint256 at such a base level is not good for modularity.

@jtimon: Sorry, this may be a pain in the ass for you as it'll mean adapting some of your other PRs, but since this is already causing indirect issues in #5995, I think it's better to get this in first.

### CODE REVIEW

**Comment by jtimon** [comment]
ut ACK

**Comment by sipa** [comment]
utACK

**Comment by jgarzik** [comment]
ut ACK

**Comment by laanwj** [comment]
#6000 !
utACK

File: src/chainparams.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ class CMainParams : public CChainParams {
         consensus.nMajorityEnforceBlockUpgrade = 750;
         consensus.nMajorityRejectBlockOutdated = 950;
         consensus.nMajorityWindow = 1000;
-        consensus.powLimit = ~arith_uint256(0) >> 32;
+        consensus.powLimit = uint256S("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
```
Comment (laanwj) [inline src/chainparams.cpp]: Nice to have these explicit.

**Pull request merged**